src/module/functions/local_function/mod.rs

```diff
@@ -247,10 +247,38 @@ impl LocalFunction {
     }
 }
 
+fn block_result_tys(
+    ctx: &ValidationContext,
+    ty: wasmparser::TypeOrFuncType,
+) -> Result<Arc<[ValType]>> {
+    match ty {
+        wasmparser::TypeOrFuncType::Type(ty) => ValType::from_wasmparser_type(ty).map(Into::into),
+        wasmparser::TypeOrFuncType::FuncType(idx) => {
+            let ty = ctx.indices.get_type(idx)?;
+            Ok(ctx.module.types.results(ty))
+        }
+    }
+}
+
+fn block_param_tys(
+    ctx: &ValidationContext,
+    ty: wasmparser::TypeOrFuncType,
+) -> Result<Arc<[ValType]>> {
+    match ty {
+        wasmparser::TypeOrFuncType::Type(_) => Ok([][..].into()),
```

Comment:
I think this `.into()` can be somewhat expensive, so I wonder if a cached version of an empty list should be stored either in `ValidationContext` or in `module.types`?

Reply:
Sure thing.